Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/repo/ContactDiaryRepository.kt

```diff
@@ -0,0 +1,41 @@
+package de.rki.coronawarnapp.contactdiary.storage.repo
+
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPerson
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryPersonEncounter
+import kotlinx.coroutines.flow.Flow
+import org.joda.time.LocalDate
+
+interface ContactDiaryRepository {
+
+    // Location
+    val locations: Flow<List<ContactDiaryLocation>>
+    suspend fun addLocation(contactDiaryLocation: ContactDiaryLocation)
+    suspend fun updateLocation(contactDiaryLocation: ContactDiaryLocation)
+    suspend fun deleteLocation(contactDiaryLocation: ContactDiaryLocation)
+    suspend fun deleteLocations(contactDiaryLocations: List<ContactDiaryLocation>)
+    suspend fun deleteAllLocations()
+
+    // Location visit
+    val locationVisits: Flow<List<ContactDiaryLocationVisit>>
+    fun locationVisitsForDate(date: LocalDate): Flow<List<ContactDiaryLocationVisit>>
+    suspend fun addLocationVisit(contactDiaryLocationVisit: ContactDiaryLocationVisit)
+    suspend fun deleteLocationVisit(contactDiaryLocationVisit: ContactDiaryLocationVisit)
+    suspend fun deleteAllLocationVisits()
+
+    // Person
+    val people: Flow<List<ContactDiaryPerson>>
+    suspend fun addPerson(contactDiaryPerson: ContactDiaryPerson)
+    suspend fun updatePerson(contactDiaryPerson: ContactDiaryPerson)
+    suspend fun deletePerson(contactDiaryPerson: ContactDiaryPerson)
+    suspend fun deletePeople(contactDiaryPeople: List<ContactDiaryPerson>)
+    suspend fun deleteAllPeople()
```

Comment:
I would strongly recommend to use `Persons` here. Grammatically speaking, that's still fine. I think just in spoken language, people is more common that persons when using plural, but I wouldn't do that in code.

Reply:
good point. will change it